app/models/friendship.rb

```diff
@@ -13,27 +13,29 @@ def request_friendship_self
   end
 
   def self.decline_friendship(friendship)
-    friendship_id = Friendship.where(user_id: friendship[:user_id], friend_id: friendship[:friend_id]).pluck(:id)
-    Friendship.destroy(friendship_id)
+    friendship_id1 = Friendship.where(user_id: friendship[:user_id], friend_id: friendship[:friend_id]).pluck(:id)
+    friendship_id2 = Friendship.where(user_id: friendship[:friend_id], friend_id: friendship[:user_id]).pluck(:id)
+    Friendship.destroy(friendship_id1)
+    Friendship.destroy(friendship_id2)
   end
 
   def self.accept_friendship(friendship)
-    friendship_id = Friendship.where(user_id: friendship[:user_id], friend_id: friendship[:friend_id]).pluck(:id)
-    f1 = Friendship.find_by(id: friendship_id)
+    friendship_id1 = Friendship.where(user_id: friendship[:user_id], friend_id: friendship[:friend_id]).pluck(:id)
+    friendship_id2 = Friendship.where(user_id: friendship[:friend_id], friend_id: friendship[:user_id]).pluck(:id)
+    f1 = Friendship.find_by(id: friendship_id1)
     f1.status = true
     f1.save
+    f2 = Friendship.find_by(id: friendship_id2)
+    f2.status = true
+    f2.save
   end
 
-  def self.undo_friendship(friendship)
-    friendship_id = Friendship.where(user_id: friendship[:user_id], friend_id: friendship[:friend_id], status: true).pluck(:id)
-    if !friendship_id.empty?
-      Friendship.destroy(friendship_id)
-      nil
-    else
-      friendship_id = Friendship.where(user_id: friendship[:friend_id], friend_id: friendship[:user_id], status: true).pluck(:id)
-      Friendship.destroy(friendship_id)
-      nil
-    end
+  def self.request_frinedship(friendship)
+    user = User.find(friendship[:user])
+    friend = User.find(friendship[:friend])
+    user.friendships.create(friend: friend, status: false)
+    friend.friendships.create(friend: user, status: nil)
+    puts
   end
 
   def self.check_friendship_status(user, current_user)
```

Comment:
**OPTIONAL**
Can be replaced by this
```ruby
# User model

# Users who needs to confirm friendship
has_many :pending_friendships, -> { where confirmed: false }, class_name: "Friendship", foreign_key: "user_id"
has_many :pending_friends, through: :pending_friendships, source: :friend

# Users who requested to be friends (needed for notifications)
has_many :inverted_friendships, -> { where confirmed: false }, class_name: "Friendship", foreign_key: "friend_id"
has_many :friend_requests, through: :inverted_friendships
```

It can simplify your code.